packages/@cdktf/provider-generator/lib/get/constructs-maker.ts

```diff
@@ -571,7 +591,9 @@ a NODE_OPTIONS variable, we won't override it. Hence, the provider generation mi
     }
 
     const jsiiTimer = logTimespan("JSII");
-    await generateJsiiLanguage(this.code, opts, this.codeMakerOutdir);
+    await generateJsiiLanguage(this.code, opts, this.codeMakerOutdir, [
+      target.isModule ? "providers/**" : "modules/**",
```

Comment:
Is this another optimization not mentioned in the PR description? Or am I missing something?

Reply:
Yeah, we have been shipping the JS code for all providers and modules in the tar.gz files for each module and provider. After this change the size of the vpc modules tar.gz file went from 12MB to 100kb and this affects the load time as well. 

This is a very simple way to achieve some benefit, unfortunately there is some hard coupling in place to get to know the path where the provider ends up and I didn’t want to start a bigger refactoring here